### PR TITLE
Feature/test-otp

### PR DESCRIPTION
**Pull request by akoidan**

### CODE REVIEW

File: src/main/java/com/icthh/xm/ms/otp/config/ApplicationProperties.java [hunk 1]
```diff
@@ -21,9 +21,12 @@
     private String kafkaSystemTopic;
     private String kafkaSystemQueue;
     private boolean timelinesEnabled;
-
+    private String specPath;
+    private String communicationUrl;
+    private String otpSenderId;
 
     private final Lep lep = new Lep();
+    private final Uaa uaa = new Uaa();
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/config/ApplicationProperties.java]: UAA config must be in tentan configuration

File: src/main/java/com/icthh/xm/ms/otp/config/ApplicationProperties.java [hunk 2]
```diff
@@ -33,4 +36,14 @@
         private TenantScriptStorage tenantScriptStorage;
         private String lepResourcePathPattern;
     }
+
+    @Getter
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/config/ApplicationProperties.java]: Remove this from app config, tenant config must be used instead

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 3]
```diff
@@ -1,21 +1,36 @@
...
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestTemplate;
 
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Optional;
+import java.net.URI;
+import java.time.Instant;
+import java.util.*;
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: remove * import

File: src/main/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResource.java [hunk 4]
```diff
@@ -9,10 +9,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResource.java]: remove * imports

File: src/main/resources/config/application.yml [hunk 5]
```diff
@@ -146,10 +146,18 @@ xm-config:
...
 
 application:
+    communication-url: http://communication
+    otp-sender-id: Vodafone
     kafka-system-topic: system_topic
     kafka-system-queue: system_queue
     tenant-ignored-path-list: /swagger-resources/configuration/ui, /management/health, /oauth/token_key, /h2-console
     timelines-enabled: true
+    spec-path: /config/tenants/XM/otp/otpspec.yml
+    uaa:
```
Comment (vkirichenko) [inline src/main/resources/config/application.yml]: remove uaa comfig from properties file

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
+package com.icthh.xm.ms.otp.security;
+import com.icthh.xm.commons.tenant.TenantContext;
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: Unused import,  indent required

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Component
+@Slf4j
+public class CommunicationService {
+
+    private final static AtomicReference atomicReference = new AtomicReference();
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: for what?

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+        Map <String, String> args,
+        Map<String, String> additionalHeaders,
+        MediaType mediaType
+    ) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(mediaType);
+
+        MultiValueMap map = new LinkedMultiValueMap();
+        for (Map.Entry<String, String> arg : args.entrySet()) {
+            map.add(arg.getKey(), arg.getValue());
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: uncheced cast, can we remove it?

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+        }
+
+        for (Map.Entry<String, String> addHeader : additionalHeaders.entrySet()) {
+            headers.set(addHeader.getKey(), addHeader.getValue());
+        }
+
+        headers.set("x-tenant", tenantContext.getContext().getTenantKey().get().getValue());
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: this header not needed

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+            uaa.getSystemAuthUrl(),
+            body,
+            headers,
+            MediaType.APPLICATION_FORM_URLENCODED
+        );
+        AuthValue auth = new AuthValue();
+        auth.accessToken = (String) response.get("access_token");
+        auth.tokenType = (String) response.get("token_type");
+        auth.expiresIn = (Integer) response.get("expires_in");
+        auth.createTokenTime = System.currentTimeMillis() / 1000;
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: Can we remove magic number?

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+            map.add(arg.getKey(), arg.getValue());
+        }
+
+        for (Map.Entry<String, String> addHeader : additionalHeaders.entrySet()) {
+            headers.set(addHeader.getKey(), addHeader.getValue());
+        }
+
+        headers.set("x-tenant", tenantContext.getContext().getTenantKey().get().getValue());
+
+        HttpEntity<MultiValueMap> request = new HttpEntity<MultiValueMap>(map, headers);
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: replace with new HttpEntity<>

File: src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java [hunk 12]
```diff
@@ -1,17 +1,26 @@
...
 import com.icthh.xm.ms.otp.service.dto.OneTimePasswordDTO;
+import com.icthh.xm.ms.otp.service.impl.OneTimePasswordServiceImpl;
 import com.icthh.xm.ms.otp.service.mapper.OneTimePasswordMapper;
 import com.icthh.xm.ms.otp.web.rest.errors.ExceptionTranslator;
-
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Ignore;
```
Comment (vkirichenko) [inline src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java]: unused import

File: src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java [hunk 13]
```diff
@@ -21,480 +30,182 @@
...
-    private static final String UPDATED_STATE_KEY = "BBBBBBBBBB";
-
-    private static final Integer DEFAULT_RETRIES = 1;
-    private static final Integer UPDATED_RETRIES = 2;
+    public static final MediaType APPLICATION_JSON_UTF8 = new MediaType(
+        MediaType.APPLICATION_JSON.getType(),
+        MediaType.APPLICATION_JSON.getSubtype(),
+        Charset.forName("utf8")
+    );
+    public static final int TTL = 600;
```
Comment (vkirichenko) [inline src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java]: change to privare for all fields

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 14]
```diff
@@ -1,21 +1,41 @@
...
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestTemplate;
 
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: unused import

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 15]
```diff
@@ -1,21 +1,41 @@
...
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Date;
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: unused import

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 16]
```diff
@@ -25,28 +45,107 @@
...
+        sendOneTimePassword(message, otp.getReceiver(), oneType.getOtpSenderId());
+        return oneTimePasswordMapper.toDto(otp);
+    }
+
+    private void sendOneTimePassword(String message, String receiver, String senderId) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(HttpHeaders.AUTHORIZATION, communicationService.getSystemToken());
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        String url = applicationProperties.getCommunicationUrl()
+            + "/tmf-api/communicationManagement/v2/communicationMessage/send";
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: use tenant config, and remove unused applicationProperties

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 17]
```diff
@@ -25,28 +45,107 @@
...
+        OneTimePassword otp = getOneTimePassword(oneTimePasswordDTO, oneType, randomPasswrd);
+
+        String message = oneType.getMessage().getEn().replaceAll("\\$\\{otp}", randomPasswrd);
+        oneTimePasswordRepository.saveAndFlush(otp);
+
+        sendOneTimePassword(message, otp.getReceiver(), oneType.getOtpSenderId());
+        return oneTimePasswordMapper.toDto(otp);
+    }
+
+    private void sendOneTimePassword(String message, String receiver, String senderId) {
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: move to communication service

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 18]
```diff
@@ -25,28 +45,107 @@
...
+        body.getReceiver().add(new Receiver(receiver, receiver));
+        RequestEntity<Object> request = new RequestEntity<>(body, headers, POST, URI.create(url));
+        loadBalancedRestTemplate.exchange(request, Object.class);
+    }
+
+    private OneTimePassword getOneTimePassword(
+        OneTimePasswordDTO oneTimePasswordDTO,
+        OtpSpec.OtpTypeSpec oneType,
+        String randomPasswrd
+    ) {
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: wrong formating
...
 getOneTimePassword(OneTimePasswordDTO oneTimePasswordDTO,
...
 String randomPasswrd) {

expected, also please check in another methods

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 19]
```diff
@@ -25,28 +45,107 @@
...
+        oneTimePassword.setReceiverTypeKey(oneType.getReceiverTypeKey());
+        oneTimePassword.setRetries(oneType.getMaxRetries());
+        oneTimePassword.setTypeKey(oneTimePasswordDTO.getTypeKey());
+        oneTimePassword.setReceiver(oneTimePasswordDTO.getReceiver());
+        oneTimePassword.setPasswordHash(sha256hex);
+        oneTimePassword.setStateKey("ACTIVE");
+        return oneTimePassword;
+    }
+
+    private OtpSpec.OtpTypeSpec getOneTypeSpec(String typeKey) {
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: move to otpSpecService

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 20]
```diff
@@ -25,28 +45,107 @@
 
     private final Logger log = LoggerFactory.getLogger(OneTimePasswordServiceImpl.class);
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: use @Slf4j lombok annotation

File: src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java (hunk 12 above)
Comment (vkirichenko) [inline src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java]: unused import

File: src/main/java/com/icthh/xm/ms/otp/domain/UaaConfig.java [hunk 21]
```diff
@@ -0,0 +1,12 @@
+package com.icthh.xm.ms.otp.domain;
+
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/domain/UaaConfig.java]: Plz, remove 1 empty line.

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.springframework.http.HttpMethod.POST;
+
+@Component
+@Slf4j
+public class CommunicationService {
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: Why communication service in security package?

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+
+        HttpEntity<MultiValueMap> request = new HttpEntity<>(map, headers);
+        log.info("Post to {} with args {}", url, args);
+        return restTemplate.postForEntity(url, request, Map.class).getBody();
+    }
+
+    public String getSystemToken() {
+
+        Map<String, String> body = new HashMap<>();
+        body.put("grant_type", "password");
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: Why not use constants?

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+        UaaConfig uaa = otpSpecService.getTenantConfig().getUaa();
+        body.put("username", uaa.getSystemUsername());
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: Why not use constants?

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+        body.put("password", uaa.getSystemPassword());
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: Why not use constants?

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 26]
```diff
@@ -0,0 +1,105 @@
...
+
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Authorization", uaa.getSystemClientToken());
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: Why not use constants?

File: src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java [hunk 27]
```diff
@@ -0,0 +1,105 @@
...
+        log.info(token);
+        return token;
+    }
+
+    public void sendOneTimePassword(String message, String receiver, String senderId) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(HttpHeaders.AUTHORIZATION, this.getSystemToken());
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        String url = otpSpecService.getTenantConfig().getCommunication().getUrl()+ "/communicationMessage/send";
+        CommunicationMessage body = new CommunicationMessage();
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/security/CommunicationService.java]: Why not use builder from lombok?

File: src/main/java/com/icthh/xm/ms/otp/service/OtpSpecService.java [hunk 28]
```diff
@@ -0,0 +1,70 @@
+package com.icthh.xm.ms.otp.service;
+
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/service/OtpSpecService.java]: Plz, remove 1 empty line.

File: src/main/java/com/icthh/xm/ms/otp/service/OtpSpecService.java [hunk 29]
```diff
@@ -0,0 +1,70 @@
...
+                this.otpSpec = mapper.readValue(config, OtpSpec.class);
+            } else if (appProps.getTenantPath().equals(updatedKey)) {
+                this.tenantConfig = mapper.readValue(config, TenantConfig.class);
+            }
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    public OtpSpec.OtpTypeSpec getOneTypeSpec(String typeKey) {
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/service/OtpSpecService.java]: `getOneTypeSpec` or `getOtpTypeSpec`. Also, check for all code.

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 30]
```diff
@@ -20,35 +24,75 @@
...
+     * @param oneTimePasswordDTO the entity to generate
      * @return the persisted entity
      */
     @Override
-    public OneTimePasswordDTO save(OneTimePasswordDTO oneTimePasswordDTO) {
-        log.debug("Request to save OneTimePassword : {}", oneTimePasswordDTO);
-        OneTimePassword oneTimePassword = oneTimePasswordMapper.toEntity(oneTimePasswordDTO);
-        oneTimePassword = oneTimePasswordRepository.save(oneTimePassword);
-        return oneTimePasswordMapper.toDto(oneTimePassword);
+    @Transactional
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: `@Transactional` also for all class. Please, remove one of them as it is redundant.

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 31]
```diff
@@ -20,35 +24,75 @@
...
+        OtpSpec.OtpTypeSpec oneType = otpSpecService.getOneTypeSpec(oneTimePasswordDTO.getTypeKey());
+
+        //generate otp
+        Generex generex = new Generex(oneType.getPattern());
+        String randomPasswrd = generex.random();
+
+        //build domain
+        OneTimePassword otp = getOneTimePassword(oneTimePasswordDTO, oneType, randomPasswrd);
+
+        String message = oneType.getMessage().getEn().replaceAll("\\$\\{otp}", randomPasswrd);
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: Please, use freemarker or another template builder.

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 32]
```diff
@@ -20,35 +24,75 @@
...
+
+
+    private OneTimePassword getOneTimePassword(OneTimePasswordDTO oneTimePasswordDTO,
+                                               OtpSpec.OtpTypeSpec oneType,
+                                               String randomPasswrd) {
+        String sha256hex = DigestUtils.sha256Hex(randomPasswrd);
+        OneTimePassword oneTimePassword = new OneTimePassword();
+        Instant startDate = Instant.now();
+        Instant endDate = startDate.plusSeconds(oneType.getTtl());
+        oneTimePassword.startDate(startDate);
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: Please, use Builder from lombok.

File: src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java [hunk 33]
```diff
@@ -22,479 +30,173 @@
...
+        tdo.setReceiver(RECEIVER);
+        tdo.setReceiverTypeKey(ReceiverTypeKey.PHONE_NUMBER);
+        tdo.setTypeKey(TYPE_KEY);
+        String requestJson = toJson(tdo);
+
+        MockHttpServletRequestBuilder postContent = post("/api/one-time-password")
+            .contentType(APPLICATION_JSON_UTF8)
+            .content(requestJson);
+        MvcResult result = restMockMvc
+            .perform(postContent)
```
Comment (ishkurko) [inline src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java]: There is no another expectation?

File: src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java [hunk 34]
```diff
@@ -22,479 +30,173 @@
...
-
-        // Validate the OneTimePassword in the database
-        List<OneTimePassword> oneTimePasswordList = oneTimePasswordRepository.findAll();
-        assertThat(oneTimePasswordList).hasSize(databaseSizeBeforeUpdate);
+        String respStr = result.getResponse().getContentAsString();
+        log.info(respStr);
+        OneTimePasswordDTO oneTimePasswordDTO = toDto(respStr, OneTimePasswordDTO.class);
+        OneTimePassword byId = oneTimePasswordRepository.findById(oneTimePasswordDTO.getId()).get();
+        long actualTtl = byId.getEndDate().toEpochMilli() - byId.getStartDate().toEpochMilli();
+        Assert.assertEquals(actualTtl / 1000, TTL);
```
Comment (ishkurko) [inline src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java]: Plz, use static import and remove `Assert` word.

File: src/main/java/com/icthh/xm/ms/otp/service/OtpSpecService.java [hunk 35]
```diff
@@ -0,0 +1,69 @@
...
+                this.tenantConfig = mapper.readValue(config, TenantConfig.class);
+            }
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    public OtpSpec.OtpTypeSpec getOtpTypeSpec(String typeKey) {
+        List<OtpSpec.OtpTypeSpec> types = this.getOtpSpec().getTypes();
+        OtpSpec.OtpTypeSpec oneType = null;
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/service/OtpSpecService.java]: `oneType` -> `otpType`

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 36]
```diff
@@ -1,54 +1,122 @@
...
-        oneTimePassword = oneTimePasswordRepository.save(oneTimePassword);
-        return oneTimePasswordMapper.toDto(oneTimePassword);
+    @Transactional
+    @SneakyThrows
+    public OneTimePasswordDTO generate(OneTimePasswordDTO oneTimePasswordDTO) {
+        log.debug("Request to generate OneTimePassword : {}", oneTimePasswordDTO);
+        OtpSpec.OtpTypeSpec oneType = otpSpecService.getOtpTypeSpec(oneTimePasswordDTO.getTypeKey());
+
+        //generate otp
+        Generex generex = new Generex(oneType.getPattern());
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: `oneType` -> `otpType`

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 37]
```diff
@@ -1,54 +1,122 @@
...
+        OneTimePassword oneTimePassword = OneTimePassword.builder()
+            .startDate(startDate)
+            .endDate(endDate)
+            .receiverTypeKey(oneType.getReceiverTypeKey())
+            .receiverTypeKey(oneType.getReceiverTypeKey())
+            .retries(oneType.getMaxRetries())
+            .typeKey(oneTimePasswordDTO.getTypeKey())
+            .receiver(oneTimePasswordDTO.getReceiver())
+            .passwordHash(sha256hex)
+            .stateKey("ACTIVE").build();
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: Why not use enum class for this?

**Pull request merged**